interface/java_binding/src/test/java/org/khronos/ktx/test/KtxTexture2Test.java

```diff
@@ -227,4 +227,126 @@ public void testCreate() {
 
         texture.destroy();
     }
+
+	@Test
+	public void testInputSwizzleBasisEx() throws IOException {
+
+		// Create RGBA pixels for an image with 32x32 pixels,
+		// filled with
+		// 8 rows of red pixels
+		// 8 rows of green pixels
+		// 8 rows of blue pixels
+		// 8 rows of white pixels
+		int sizeX = 32;
+		int sizeY = 32;
+		byte[] rgba = new byte[sizeX * sizeY * 4];
+		fillRows(rgba, sizeX, sizeY, 0, 8, 255, 0, 0, 255); // Red
+		fillRows(rgba, sizeX, sizeY, 8, 16, 0, 255, 0, 255); // Green
+		fillRows(rgba, sizeX, sizeY, 16, 24, 0, 0, 255, 255); // Blue
+		fillRows(rgba, sizeX, sizeY, 24, 32, 255, 255, 255, 255); // White
+
+		// Create a texture and fill it with the RGBA pixel data
+		KtxTextureCreateInfo info = new KtxTextureCreateInfo();
+		info.setBaseWidth(sizeX);
+		info.setBaseHeight(sizeY);
+		info.setVkFormat(VkFormat.VK_FORMAT_R8G8B8A8_SRGB);
+		KtxTexture2 t = KtxTexture2.create(info, KtxCreateStorage.ALLOC);
+		t.setImageFromMemory(0, 0, 0, rgba);
+
+		// Apply basis compression with an input swizzle, BRGA, so that
+		// the former B channel becomes the R channel
+		// the former R channel becomes the G channel
+		// the former G channel becomes the B channel
+		// the former A channel remains the A channel
+		KtxBasisParams p = new KtxBasisParams();
+		p.setUastc(false);
+		p.setInputSwizzle(new char[] { 'b', 'r', 'g', 'a' });
+		t.compressBasisEx(p);
+
+		t.destroy();
+	}
+
+	@Test
+	public void testSupercompressionZstd() throws IOException {
+
+		int sizeX = 32;
+		int sizeY = 32;
+
+		// Create a dummy texture
+		KtxTextureCreateInfo info = new KtxTextureCreateInfo();
+		info.setBaseWidth(sizeX);
+		info.setBaseHeight(sizeY);
+		info.setVkFormat(VkFormat.VK_FORMAT_R8G8B8A8_SRGB);
+		KtxTexture2 t = KtxTexture2.create(info, KtxCreateStorage.ALLOC);
+		byte[] rgba = new byte[sizeX * sizeY * 4];
+		t.setImageFromMemory(0, 0, 0, rgba);
+
+		// Apply default UASTC compression
+		KtxBasisParams p = new KtxBasisParams();
+		p.setUastc(true);
+		t.compressBasisEx(p);
+
+		// The supercompression scheme should be NONE here
+		int scBefore = t.getSupercompressionScheme();
+		assertEquals(KtxSupercmpScheme.NONE, scBefore);
+
+		// Apply Zstd compression
+		t.deflateZstd(10);
+
+		// The supercompression scheme should now be ZSTD
+		int scAfter = t.getSupercompressionScheme();
+		assertEquals(KtxSupercmpScheme.ZSTD, scAfter);
+
```

Comment:
I suggest also inflating the texture again and comparing its data with `rgba`. Does `new` in java 0 initialize the memory or will it be random data.

If the UASTC compression causes this compare to fail, you can skip the `compressBasisEx` step. It is perfectly valid to deflate uncompressed textures.

---- EDIT ----

The native inflate functions are not exposed in the public API. Inflation is always done when a file is opened. (I did not think about something like Java testing when I made that choice. Native tests can always link to the function.) To make my suggestion work you will have to call t.writeToMemory then CreateFromMemory but didn't you tell me those functions aren't exposed in JNI. They need to be.

Reply:
I know that it would be _better_ to compare the data, to have a more thorough test. And this is **ONLY** Zstd (lossless), so I assume that it will not suffer from the ~"slight inaccuracies" that may appear in other comparisons, like the one described in https://github.com/KhronosGroup/KTX-Software/pull/876#discussion_r1537923784 . 

I could spend dozens of hours with ~"improving things". This includes the tests, and even the _existing_ tests (which, in the JNI binding layer, are ... a bit shallow, to be honest). But I took a note of that as well, and will try to tackle that as part of the overall "improvement"...

Reply:
There won't be any non-determinism or inaccuracies with Zstd.